lib/metrics/registry.js

```diff
@@ -21,15 +22,10 @@ class MetricsRegistry extends SelfReportingMetricsRegistry {
     const options = Object.assign({}, defaultReporterOptions, reporterOptions)
     const reporter = new MetricsReporter(transport, options)
     super(reporter, registryOptions)
-    if (options.enabled) createSystemMetrics(this, options.defaultReportingIntervalInSeconds)
-  }
-
-  shutdown () {
-    if (this.collector) {
-      this.collector.stop()
-    }
-
-    return super.shutdown()
+    this._registry.collectors = [
+      createSystemMetrics(this),
+      createRuntimeMetrics(this)
+    ].filter(v => !!v)
```

Comment:
It's filtered because only Linux has a stats collector for system metrics. On the generic platform it has no collector so that would be an empty element in the array which makes the map to `collect()` calls in the reporter break.

Reply:
I see. Maybe it's just me, but I didn't find it that obvious, especially because this is behind a few levels of abstraction. So a comment about this would still be nice I think

Reply:
I just changed how it works so the collector is registered by the code that instantiates the collector.